src/crystal/system/unix/process.cr

```diff
@@ -0,0 +1,243 @@
+require "c/signal"
+require "c/stdlib"
+require "c/sys/resource"
+require "c/unistd"
+
+class Process
+  protected def open_system
+  end
+
+  protected def wait_system
+    Crystal::SignalChildHandler.wait(@waitpid, @pid)
+  end
+
+  protected def self.exit_system(status = 0) : NoReturn
+    LibC.exit(status)
+  end
+
+  protected def self.pid_system : Int64
+    LibC.getpid.to_i64
+  end
+
+  protected def self.ppid_system : Int64
+    LibC.getppid.to_i64
+  end
+
+  protected def exists_system?
+    ret = LibC.kill(@pid.to_i32, 0)
```

Comment:
It should be a class method `self.exists_system(pid)` that could be called directly from `Process.exists?(pid)` for example, avoiding the instantiation of a Process object.

Also don't manually cast values to LibC calls: the compiler does it automatically.